### DISCUSSION TITLE
Autofocus first option

### DISCUSSION THREAD

**Discussion by lcamero-cip** [brianvoe/slim-select#431] [Q&A]
Hi!

Is there a way to make the SlimSelect dropdown autofocus the first option on the list when the element is focused or when you use the search bar?

Right now you can type in the search bar and hit the down arrow key and it will focus the first option. However, it is an extra key press. 

It does not seem to lose focus when you use the keyboard to navigate so it feels like focusing the first item would be possible without losing the search bar.

![image](https://user-images.githubusercontent.com/98397407/214612813-c62aebf7-37d6-4a67-b668-e213085e8dc0.png)

Is this supported by some setting? I don't see a reference of it in the settings documentation if it exists.

Thank you!

**Comment by giopk** [comment]
Hi Brian, 
This should become a standard behaviour as it's a typical use case when users need to fill repetitive forms and mostly use keyboard navigation. When the options are Cat and Dog for example users who do a lot of data entry expect to type D and tab and get Dog whereas now they  get nothing or the previously selected option

**Reply by JakobBernoulli** [reply]
Wishing the same. Has pr been done?

**Reply by brianvoe** [reply]
Ill work on it and get it released

**Reply by JakobBernoulli** [reply]
Thank you very much for your reply and the upcoming integration. I would like to take this opportunity to congratulate you on this great tool. I have tried many alternatives, but it was your solution that fully satisfied my needs. Very good work. Thank you very much.

**Reply by brianvoe** [reply]
Your appreciation is the reason I will be doing this. Thank you.

**Comment by brianvoe** [comment]
submit a pr that on enter auto selects first value.
